integration/system/stats_test.go

```diff
@@ -259,53 +276,45 @@ func jsonBlobHasGTE121NetworkStats(blob map[string]interface{}) bool {
 	return true
 }
 
-func (s *DockerAPISuite) TestAPIStatsContainerNotFound(c *testing.T) {
-	testRequires(c, DaemonIsLinux)
-	apiClient, err := client.NewClientWithOpts(client.FromEnv)
-	assert.NilError(c, err)
-	defer apiClient.Close()
+func TestAPIStatsNoStreamGetCpu(t *testing.T) {
+	skip.If(t, RuntimeIsWindowsContainerd(), "FIXME: Broken on Windows + containerd combination")
 
-	expected := "No such container: nonexistent"
+	ctx := setupTest(t)
+	apiClient := testEnv.APIClient()
 
-	_, err = apiClient.ContainerStats(testutil.GetContext(c), "nonexistent", true)
-	assert.ErrorContains(c, err, expected)
-	_, err = apiClient.ContainerStats(testutil.GetContext(c), "nonexistent", false)
-	assert.ErrorContains(c, err, expected)
-}
+	cID := container.Run(ctx, t, apiClient, container.WithImage("busybox"), container.WithCmd("/bin/sh", "-c", "'while true;usleep 100; do echo 'Hello'; done'"))
+	time.Sleep(300 * time.Microsecond) // Necessary to prevent windows flakyness
```

Comment:
Should we poll the API until we get the desired response?
I don't think this tiny sleep is buying us much.

Reply:
Ugh... don't know why github put this over here...
But in any case, looks like the test failed:

> === FAIL: github.com/docker/docker/integration/system TestAPIStatsNoStreamGetCpu (2.12s)
    stats_test.go:319: assertion failed: cpuPercent is 0.0: docker stats with no-stream get cpu usage failed: was 0